luma/lcd/device.py

```diff
@@ -88,6 +89,124 @@ def contrast(self, value):
+        self.command(0xB2, 0x01, 0x2C, 0x2D)    # frame rate control: idle mode
+        self.command(0xB3, 0x01, 0x2C, 0x2D,    # frame rate control: partial mode dot inversion mode
+                     0x01, 0x2C, 0x2D)          # frame rate control: line inversion mode
+        self.command(0xB4, 0x07)                # display inversion: none
+        self.command(0xC0, 0xA2, 0x02, 0x84)    # power control 1: -4.6V auto mode
+        self.command(0xC1, 0xC5)                # power control 2: VGH
+        self.command(0xC2, 0x0A, 0x00)          # power control 3: OpAmp current small, boost freq
+        self.command(0xC3, 0x8A, 0x2A)          # power control 4: BCLK/2, Opamp current small & Medium low
+        self.command(0xC4, 0x8A, 0xEE)          # power control 5: partial mode/full-color
+        self.command(0xC5, 0x0E)                # VCOM Control 1
+        self.command(0x36, 0x60 | order)        # memory data access control
+        self.command(0x20)                      # display inversion off
+        self.command(0x3A, 0x06)                # interface pixel format
+        self.command(0x13)                      # partial off (normal)
+        self.command(0xE0,                      # gamma adjustment (+ polarity)
+                     0x0F, 0x1A, 0x0F, 0x18, 0x2F, 0x28, 0x20, 0x22,
+                     0x1F, 0x1B, 0x23, 0x37, 0x00, 0x07, 0x02, 0x10)
+        self.command(0xE1,                      # gamma adjustment (- polarity)
+                     0x0F, 0x1B, 0x0F, 0x17, 0x33, 0x2C, 0x29, 0x2E,
+                     0x30, 0x30, 0x39, 0x3F, 0x00, 0x07, 0x03, 0x10)
+
+        self.clear()
+        self.show()
+
+    def display(self, image):
+        """
+        Renders a 24-bit RGB image to the ST7735 LCD display
+
+        :param image: the image to render
+        :type image: PIL.Image.Image
+        """
+        assert(image.mode == self.mode)
+        assert(image.size == self.size)
+
+        image = self.preprocess(image)
+
+        if self.framebuffer.redraw_required(image):
+            left, top, right, bottom = self.framebuffer.bounding_box
+            width = right - left
+            height = bottom - top
+
+            self.command(0x2A, left >> 8, left & 0xFF, (right - 1) >> 8, (right - 1) & 0xFF)     # Set column addr
+            self.command(0x2B, top >> 8, top & 0xFF, (bottom - 1) >> 8, (bottom - 1) & 0xFF)     # Set row addr
+            self.command(0x2C)                                                                   # Memory write
+
+            i = 0
+            buf = bytearray(width * height * 3)
+            for r, g, b in self.framebuffer.getdata():
+                if not(r == g == b == 0):
+                    # 262K format
+                    buf[i] = r
+                    buf[i + 1] = g
+                    buf[i + 2] = b
+                i += 3
+
+            self.data(list(buf))
+
+    def contrast(self, level):
+        """
+        NOT SUPPORTED
```

Comment:
Need to override the default implementation with a `pass` so it doesn't do anything. The only way to set the contrast for these LCDs is to PWM the backlight... something that the backlight class doesn't do just yet.